app/controllers/users_controller.rb

```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = 'Object successfully created'
+      redirect_to new_user_path
+    else
+      flash[:error] = 'Something went wrong'
+      render 'new'
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
```

Comment:
## Implement `update` action

* [ ] finds user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
* [ ] `redirect_to edit_user_path` if user was saved
* [ ] `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

Reply:
We have implemented this, however, #update doesn't render since we did not create a show.html.erb form, we will leave it as is since it's not a requirement.

Reply:
_"Lastly, what if we could validate the email format by using either a regular expression or a built-in URI method?"_
Also, regarding the email validation, there is a RegEx existing and it is currently working. See **app/models/user.rb** **line 5**
```
validates :email,
            presence: true,
            uniqueness: true,
            format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
```